src/features/fee-row/fee-row.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { FiInfo } from 'react-icons/fi';
+import { useField } from 'formik';
+import { Box, color, Flex, Stack, Tooltip } from '@stacks/ui';
+
+import { stacksValue } from '@common/stacks-utils';
+import { LoadingRectangle } from '@components/loading-rectangle';
+import { SpaceBetween } from '@components/space-between';
+import { Caption } from '@components/typography';
+import { useFeeRateState, useFeeState } from '@store/transactions/fees.hooks';
+
+import { TransactionFee } from './components/transaction-fee';
+import { FeeEstimateItem } from './components/fee-estimate-item';
+import { FeeEstimateSelect } from './components/fee-estimate-select';
+import { CustomFeeField } from './components/custom-fee-field';
+import { SendFormErrorMessages } from '@common/error-messages';
+import { CustomFeeError } from './components/custom-fee-error';
+import { FeeEstimation } from '@models/fees-types';
+
+interface FeeRowProps {
+  feeEstimations: FeeEstimation[] | null;
+}
+
+// Fallback component
+function FeeRowFallback(): JSX.Element {
+  return (
+    <SpaceBetween>
+      <Caption>Fees</Caption>
+      <LoadingRectangle width="50px" height="10px" />
+    </SpaceBetween>
+  );
+}
+
+export function FeeRow(props: FeeRowProps): JSX.Element {
+  const { feeEstimations } = props;
+  const [, meta, helpers] = useField('fee');
+  const [isOpen, setIsOpen] = useState(false);
+  const [fee, setFee] = useFeeState();
+  const [, setFeeRate] = useFeeRateState();
+  const hasFeeEstimations = feeEstimations && !!feeEstimations.length;
+  const [selected, setSelected] = useState(hasFeeEstimations ? 1 : 3);
```

Comment:
What are 1 & 3? 🤷🏼  If these are referencing some hardcoded array length, maybe an enum could help readability?

Reply:
Yep, makes sense, prob rushed thru that. That is the length of the estimations array for the query + the custom field. I'll make that more clear. 👍